utils/structs/structs_reflection.go

```diff
@@ -44,36 +44,103 @@ func FindField(pointerToAField interface{}, pointerToAStruct interface{}) (field
 }
 
 // ListExportedFields returns all fields of a structure that starts wit uppercase letter
-func ListExportedFields(val interface{}) []*reflect.StructField {
+func ListExportedFields(val interface{}, predicates ...ExportedPredicate) []*reflect.StructField {
 	valType := reflect.Indirect(reflect.ValueOf(val)).Type()
 	len := valType.NumField()
 	ret := []*reflect.StructField{}
 	for i := 0; i < len; i++ {
-		field := valType.Field(i)
-		if field.Name[0] == strings.ToUpper(string(field.Name[0]))[0] {
-			// if exported
-			ret = append(ret, &field)
+		structField := valType.Field(i)
+
+		if FieldExported(&structField, predicates...) {
+			ret = append(ret, &structField)
 		}
 	}
 
 	return ret
 }
 
 // ListExportedFieldsWithVals returns all fields of a structure that starts wit uppercase letter with values
-func ListExportedFieldsWithVals(val interface{}) (fields []*reflect.StructField, values []interface{}) {
+func ListExportedFieldsWithVals(val interface{}, predicates ...ExportedPredicate) (fields []*reflect.StructField, values []interface{}) {
 	valRefl := reflect.Indirect(reflect.ValueOf(val))
 	valType := valRefl.Type()
 	len := valType.NumField()
 	fields = []*reflect.StructField{}
 	values = []interface{}{}
 	for i := 0; i < len; i++ {
-		field := valType.Field(i)
-		if field.Name[0] == strings.ToUpper(string(field.Name[0]))[0] {
+		structField := valType.Field(i)
+
+		if FieldExported(&structField, predicates...) {
 			// if exported
-			fields = append(fields, &field)
+			fields = append(fields, &structField)
 			values = append(values, valRefl.Field(i).Interface())
 		}
 	}
 
 	return fields, values
 }
+
+// ExportedPredicate defines a callback (used in func FieldExported)
+type ExportedPredicate func(field *reflect.StructField) bool
+
+// FieldExported returns true if field name starts with uppercase
+func FieldExported(field *reflect.StructField, predicates ...ExportedPredicate) (exported bool) {
+	if field.Name[0] == strings.ToUpper(string(field.Name[0]))[0] {
+		expPredic := true
+		for _, predicate := range predicates {
+			if !predicate(field) {
+				expPredic = false
+				break
+			}
+		}
+
+		return expPredic
+	}
+
+	return false
+}
+
+// ListExportedFieldsPtrs iterates struct fields and return slice of pointers to field values
+func ListExportedFieldsPtrs(val interface{}, predicates ...ExportedPredicate) []interface{} {
+	rVal := reflect.Indirect(reflect.ValueOf(val))
+	ptrs := []interface{}{}
+	for i := 0; i < rVal.NumField(); i++ {
+		field := rVal.Field(i)
+		structField := rVal.Type().Field(i)
+		if !FieldExported(&structField) {
```

Comment:
What about: if !FieldExported(&structField, predicates...) {
Then we can avoid the for cycle below to appear in this function as well as in FieldExported().